simplesat/transaction.py

```diff
@@ -113,23 +121,40 @@ def _safe_operations(self, pool, decisions, installed_package_ids):
                       package_id not in installed_package_ids):
                     installs.append(package_id)
 
-        # Removals should happen top down
-        for package_id in reversed(removals):
-            package = pool.id_to_package(package_id)
-            operations.append(RemoveOperation(package))
+        install_operations = []
+        remove_operations = []
 
         # Installations should happen bottom up
         for package_id in installs:
-            package = pool.id_to_package(package_id)
-            operations.append(InstallOperation(package))
+            # The solver may suggest removing a currently installed package and
+            # installing another version of that same package from a different
```

Comment:
Oooh, interesting. That opens up a whole scope of questions. How does the user specify which `weekly_reports` package they want? How do other packages refer to it when specifying dependencies and conflicts? Can you get both `weekly_reports`? Do we really want to have to handle all of that?